modules/service/src/main/scala/lucuma/odb/graphql/mapping/MutationMapping.scala

```diff
@@ -739,6 +760,33 @@ trait MutationMapping[F[_]] extends Predicates[F] {
       }
     }
 
+  private lazy val UpdateProgramUsers =
+    MutationField("updateProgramUsers", UpdateProgramUsersInput.binding(Path.from(ProgramUserType))) { (input, child) =>
+      services.useTransactionally {
+        val filterPredicate = and(List(
+          Predicates.programUser.program.isWritableBy(user),
```

Comment:
Here we check the first time that the user has write access, when we figure out which program users we will attempt to update.  The `isWritableBy` predicate needs to be updated to work for read-only CoIs and for NGO users.

Reply:
Yeah all of these things are on my very long list of things to review once the obs status stuff is working.